platform/security/sts/security-sts-ldapclaimshandler/src/main/java/ddf/security/sts/claimsHandler/AttributeMapLoader.java

```diff
@@ -94,6 +96,37 @@ public static String getUser(Principal principal) {
         return user;
     }
 
+    /**
+     * Determines the base DN for the provided {@code Principal}.
+     *
+     * If an X500 {@code Principal} is provided, its base DN is extracted; however, if it is empty
+     * then then provided {@code defaultBaseDN} is used instead.
+     *
+     * If any other type of {@code Principal} is provided, the {@code defaultBaseDN} is used.
+     *
+     * @param principal the Principal to check
+     * @param defaultBaseDN the default DN to fall back to
+     * @return the base DN
+     */
+    public static String getBaseDN(Principal principal, String defaultBaseDN) {
+        String baseDN = null;
+        if (principal instanceof X500Principal) {
+            baseDN = Pattern.compile(",")
```

Comment:
You might be able to use X500Name from bouncy castle to get this. See: https://github.com/codice/ddf/blob/master/platform/security/platform-security-core-api/src/main/java/ddf/security/SubjectUtils.java#L146